src/common/classes/auto.h

```diff
@@ -168,6 +168,16 @@ class AutoPtr
 };
 
 
+template <typename ID>
```

Comment:
@AlexPeshkoff maybe we can use C++11's template ... using here?

Reply:
@asfernandes On my mind what is needed here is template template parameter, i.e.

template<typename Where, template<typename> class Clear = SimpleDelete>
class AutoPtr { .................

but I'm not sure have we decided to use them in FB4 or not.